examples/step-58/doc/intro.dox

```diff
@@ -0,0 +1,840 @@
+
+At first glance, the equations appear to be parabolic and similar to
+the heat equation (see step-26) as there is only a single time
+derivative and two spatial derivatives. But this is misleading and
+more easily seen if we assume for a moment that the potential $V=0$
+and $\kappa=0$. Then we have the equation
+@f{align*}{
+  - i \frac{\partial \psi}{\partial t}
+  - \frac 12 \Delta \psi
+  &= 0.
+@f}
+If we separate the solution into real and imaginary parts, $\psi=v+iw$,
+with $v=\textrm{Re}\;\psi,\; w=\textrm{Im}\;\psi$,
+then we can split the one equation into its real and imaginary parts
+in the same way as we did in step-29:
+@f{align*}{
+  \frac{\partial w}{\partial t}
+  - \frac 12 \Delta v
+  &= 0,
+  \\
+  -\frac{\partial v}{\partial t}
+  - \frac 12 \Delta w
+  &= 0.
+@f}
+Not surprisingly, the factor $i$ in front of the time derivative
+couples the real and imaginary parts of the equation. If we want to
+understand this equation further, take the time derivative of one of
+the equations, say
+@f{align*}{
+  \frac{\partial^2 w}{\partial t^2}
+  - \frac 12 \Delta \frac{\partial v}{\partial t}
+  &= 0,
+@f}
+(where we have assumed that, at least in some formal sense, we can
+commute the spatial and temporal derivatives), and then insert the
+other equation into it:
+@f{align*}{
+  \frac{\partial^2 w}{\partial t^2}
+  + \frac 14 \Delta^2 w
+  &= 0.
+@f}
+This equation is hyperbolic and similar in character to the wave
+equation. (This will also be obvious if you look at the video
+in the "Results" section of this program.) Furthermore, we could
+have arrived at the same equation for $v$ as well.
+Consequently, a better assumption for the NLSE is to think of
+it as a hyperbolic, wave-propagation equation than as a diffusion
+equation such as the heat equation. (You may wonder whether it is
+correct that the operator $\Delta^2$ appears with a positive sign
+whereas in the wave equation, $\Delta$ has a negative sign. This is
+indeed correct: After multiplying by a test function and integrating
+by parts, we want to come out with a positive (semi-)definite
+form. So, from $-\Delta u$ we obtain $+(\nabla v,\nabla u)$. Likewise,
+after integrating by parts twice, we obtain from $+\Delta^2 u$ the
+form $+(\Delta v,\Delta u)$. In both cases do we get the desired positive
+sign.)
+
+The real NLSE, of course, also has the terms $V\psi$ and
+$\kappa|\psi|^2\psi$. However, these are of lower order in the spatial
+derivatives, and while they are obviously important, they do not
+change the character of the equation.
+
+In any case, the purpose of this discussion is to figure out
+what time stepping scheme might be appropriate for the equation. The
+conclusions is that, as a hyperbolic-kind of equation, we need to
+choose a time step that satisfies a CFL-type condition. If we were to
+use an explicit (which we will not), we would have to investigate
+the eigenvalues of the spatial operator. If you followed the discussions
```

Comment:
It's really about the eigenvalues of the matrix that corresponds to the spatial operator. I fixed this. But in general, the approach is the same for operators: You try to find, for example, those values `lambda` so that
```
  -Delta u(x) = lambda u(x)
```
The same thing can be done for any differential operator.